### PR TITLE
refactor DenoPermissions.check_net & resolve_addr

### LINKED ISSUE #3170: refactor: Rewrite DenoPermissions.check_net to (host, port)
https://github.com/denoland/deno/blob/f1fa3bd792712a98f7fc8566a3bf4e1c7d9901b1/cli/ops/net.rs#L73-L77

This is a good first issue in Rust

### PR DESCRIPTION
**Pull request by manyuanrong**
Closes #3170
cc @ry

### CODE REVIEW

File: cli/ops/tls.rs [hunk 1]
```diff
@@ -55,23 +55,20 @@ pub fn op_dial_tls(
...
-  let address = format!("{}:{}", args.hostname, args.port);
   let cert_file = args.cert_file;
 
-  state.check_net(&address)?;
   if let Some(path) = cert_file.clone() {
     state.check_read(&path)?;
   }
 
-  let mut domain = args.hostname;
+  state.check_net(&args.hostname, args.port)?;
```
Comment (bartlomieju) [inline cli/ops/tls.rs]: I suggest you keep the same order of checks (ie. move this `check_state` above `check_read`)

File: cli/permissions.rs [hunk 2]
```diff
@@ -243,27 +243,25 @@ impl DenoPermissions {
...
       PermissionAccessorState::Allow => {
         self.log_perm_access(msg);
         Ok(())
       }
       state => {
-        let parts = host_and_port.split(':').collect::<Vec<&str>>();
-        if match parts.len() {
-          2 => {
-            if self.net_whitelist.contains(parts[0]) {
+        if match port {
```
Comment (bartlomieju) [inline cli/permissions.rs]: This whole if might be skipped IMHO, this should be sufficient:
```rust
if self.net_whitelist.contains(hostname) {
  true
} else {
    self
     .net_whitelist
     .contains(&format!("{}:{}", hostname, port))
}
```

File: cli/resolve_addr.rs [hunk 3]
```diff
@@ -9,19 +8,21 @@ use std::net::ToSocketAddrs;
 
 /// Go-style network address parsing. Returns a future.
```
Comment (bartlomieju) [inline cli/resolve_addr.rs]: This comment is no longer valid :)

**Comment by manyuanrong** [comment]
@bartlomieju ready for review

**Pull request merged**